### ISSUE TITLE
Move core modules to @textlint namespace

### ISSUE THREAD

**Issue by azu** [textlint/textlint#455]
npm recommened to use scoped.

> If you are prevented from publishing a package with a name too similar to an existing package, the easiest way to find a unique name is to use your scope. You can scope packages to your npm account by putting @ + your npm user name in front of the name. For example, my npm user name is ceejbot, so my scope is @ceejbot.
> http://blog.npmjs.org/post/168978377570/new-package-moniker-rules

We already have [@textlint](https://www.npmjs.com/org/textlint "@textlint") scope.

### Pros

- Make explicit

For example, [markdown-to-ast](https://github.com/textlint/textlint/tree/master/packages/markdown-to-ast "markdown-to-ast") is confused name.

### Cons

- Name chaning!

But, almost pacakges are only used in textlint.

- [textlint/packages/](https://github.com/textlint/textlint/tree/master/packages "textlint/packages/")

## Migration pacakges

| PR    | Before                     | After                                |
| :-----  | -------------------------- | ------------------------------------ |
| #473 | `txt-to-ast`               | `@textlint/text-to-ast`              |
| #460 | `markdown-to-ast`          | `@textlint/markdown-to-ast`          |
| #483 | `textlint-ast-test`        | `@textlint/ast-tester`               |
| #484 | `txt-ast-traverse`         | `@textlint/ast-traverse`             |
| #482 | `textlint-fixer-formatter` | `@textlint/fixer-formatter`          |
| #474 | `textlint-formatter`       | `@textlint/linter-formatter`         |
| #485 | `textlint-plugin-markdown` | `@textlint/textlint-plugin-markdown` |
| #486 | `textlint-plugin-text`     | `@textlint/textlint-plugin-text`     |

<details>
<summary> Migration Packages List </summary>

- [x] txt-to-ast -> `@textlint/text-to-ast` #473 
- [x] markdown-to-ast -> `@textlint/markdown-to-ast` #460 
- [x] textlint-ast-test -> `@textlint/ast-tester` #483
  - Add `er` is good
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [x] txt-ast-traverse -> `@textlint/ast-traverse` #484
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [x] textlint-fixer-formatter -> `@textlint/fixer-formatter` #482
- [x] textlint-formatter -> `@textlint/linter-formatter` #474
- [x] textlint-plugin-markdown -> `@textlint/textlint-plugin-markdown` #485
- [ ] textlint-plugin-text -> `@textlint/textlint-plugin-text` #486
  - `textlint-plugin` prefix is needed
  - [Package name convention](https://github.com/textlint/textlint/blob/master/docs/plugin.md#package-name-convention "Package name convention")

</details>

## No migration packages

These packages are used by external user.

- textlint
   - textlint it-self!
- gulp-textlint
   - textlint user use it
- textlint-tester
  - textlint rule creator use it

## ℹ️ Migration information for user

If you use someone of Migration pacakges, please move to scoped version.

For example, If you use `markdown-to-ast`, you migrate from `markdown-to-ast` to `@textlint/markdown-to-ast`. 

```
npm uninstall markdown-to-ast
npm install @textlint/markdown-to-ast
```

And, you should change `require("markdown-to-ast")` to `require("@textlint/markdown-to-ast")`.

-----

## How to migrate?

Migrate to `@textlint`.
<del>Maybe, [transfer-github-owner](https://github.com/azu/transfer-github-owner "transfer-github-owner") help this.</del>

Example: #473 #460 

1.  Move `<name>` to  `@textlint/<name>` directory 710a09e 
2. change usage of `<name>` to `@textlint/<name>` 5a12178 
  - And update `yarn.lock` by `yarn bootstrap`
3. Update core package table ec78494 
  - 💔 textlint-rule-no-dead-link found it as dead link
4. Add "publishConfig" 929b575 
  - `"publishConfig": { "access": "public" }` config is needed when you publish [Scoped packages](https://docs.npmjs.com/misc/scope "Scoped packages") at first time
  - Alt. `npm publish --access public` is anthor option
  - But, `publishConfig` is more reasonable.
5. `npm publish` it as `@textlint/<name>` from local
  - ❤️ This fix dead link of "Update core package table"
  - It create https://www.npmjs.com/package/@textlint/markdown-to-ast on npm.
6. Create PR

Finally, deprecate old pacakge:

```
npm deprecate <old-pacakge> "See https://github.com/textlint/textlint/issues/455"
```

**Comment by 0x6b** [comment]
@azu, I have a question; do we migrate `textlint-*` to `@textlint/textlint-*`, or to `@textlint/*`? e.g. `@textlint/textlint-formatter` or `@textlint/formatter`?

Since `textlint-` prefix is for name space, use it under scoped environment looks redundant for me.

**Comment by azu** [comment]
@0x6b Good question!

I've updated migration name.

- [ ] txt-to-ast -> `@textlint/txt-to-ast`
- [ ] markdown-to-ast -> `@textlint/markdown-to-ast`
- [ ] textlint-ast-test -> `@textlint/ast-tester`
  - Add `er` is good
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [ ] textlint-fixer-formatter -> `@textlint/fixer-formatter`
- [ ] textlint-formatter -> `@textlint/lint-formatter`
- [ ] txt-ast-traverse -> `@textlint/ast-traverse`
  - Align with [@textlint/ast-node-types](https://github.com/textlint/textlint/tree/master/packages/%40textlint/ast-node-types "@textlint/ast-node-types")
- [ ] textlint-plugin-markdown -> `@textlint/textlint-plugin-markdown`
- [ ] textlint-plugin-text -> `@textlint/textlint-plugin-text`
  - `textlint-plugin` prefix is needed
  - [Package name convention](https://github.com/textlint/textlint/blob/master/docs/plugin.md#package-name-convention "Package name convention")

**Comment by 0x6b** [comment]
@azu, thank you for the clarification. It's good move.

> `txt-to-ast` -> `@textlint/txt-to-ast` or `@textlint/text-to-ast`

+1 for `@textlint/text-to-ast`. Mixing text with txt in a package name is confusing and error–prone?

**Comment by azu** [comment]
I decided that rename `txt-to-ast` to `@textlint/text-to-ast`

**Comment by azu** [comment]
> no-dead-link rule fails because migrated package has new URL e.g. https://www.npmjs.com/package/@textlint/formatter

Yes! I've tried that move `markdown-to-ast` to `@textlint/linter-formatter` #460
#460's step 5(`npm publish` from local) is needed to pass CI.
In other words,  step 5 create placeholder(temporary) page <https://www.npmjs.com/package/@textlint/markdown-to-ast>

It is better that complete migration task in a single PR :)

I've added @0x6b to npm [@textlint](https://www.npmjs.com/org/textlint/members "@textlint") member.
So, You can publish new package to `@textlint` scope.
In other words, you can do step5.

**Comment by azu** [comment]
@Leko I want to invite you to npm [@textlint](https://www.npmjs.com/org/textlint/members "@textlint") member.
Your npm account is [lekoleko](https://www.npmjs.com/~lekoleko "lekoleko")?

**Comment by 0x6b** [comment]
@azu, thank you for adding me to the npm organization!
As for new name for `textlint-formatter`, I have to rework on local branch...anyway thank you for pointing out.

**Comment by 0x6b** [comment]
@azu, while refactoring `textlint-plugin-markdown` to `@textlint/textlint-plugin-markdown`, `textlint-engine-test` was failed with following error:

```
  1) textlint-engine-test
       #loadPlugin
         when Plugin has not rules
           should not throw Error:
     ReferenceError: Failed to load textlint's plugin module: "markdown" is not found.
```

**Comment by azu** [comment]
@0x6b 
https://github.com/textlint/textlint/blob/442c297c60e77de55f07fea7c75d71ac05a6a03a/packages/textlint/test/engine/textlint-engine-test.ts#L84

Probably, you need to change plugin name.
This plugin name is shortcut of pacakge name(`@textlint/textlint-plugin-markdown`)

- `markdown` -> `textlint-plugin-markdown`
- `@textlint/markdown` -> `textlint/textlint-plugin-markdown`

```diff
- const engine = new TextLintEngine({ plugins: ["markdown"] });
+ const engine = new TextLintEngine({ plugins: ["@textlint/markdown"] });
```

**Comment by 0x6b** [comment]
@azu, you are right! (as always)
`textlint-module-resolver` already take account this path. Impressed. Thank you.

https://github.com/textlint/textlint/blob/a144570246d0ec89cbe370f80b5e0706e5c0647a/packages/textlint/src/engine/textlint-module-resolver.ts#L26

**Comment by 0x6b** [comment]
```sh
$ for package in txt-to-ast markdown-to-ast textlint-ast-test txt-ast-traverse textlint-fixer-formatter textlint-formatter textlint-plugin-markdown textlint-plugin-text; do echo -n "$package: "; npm info --json $package | jq .deprecated; done
txt-to-ast: "See https://github.com/textlint/textlint/issues/455"
markdown-to-ast: "See https://github.com/textlint/textlint/issues/455"
textlint-ast-test: "See https://github.com/textlint/textlint/issues/455"
txt-ast-traverse: "See https://github.com/textlint/textlint/issues/455"
textlint-fixer-formatter: "See https://github.com/textlint/textlint/issues/455"
textlint-formatter: "See https://github.com/textlint/textlint/issues/455"
textlint-plugin-markdown: "See https://github.com/textlint/textlint/issues/455"
textlint-plugin-text: "See https://github.com/textlint/textlint/issues/455"
```

Now we can go to another set of refactoring such as TypeScript-ify or Jest-ing (#451).

**Comment by 0x6b** [comment]
Found that `npm view <package> deprecated` is simpler than `npm info --json <package> | jq .deprecated`.

```sh
$ for package in txt-to-ast markdown-to-ast textlint-ast-test txt-ast-traverse textlint-fixer-formatter textlint-formatter textlint-plugin-markdown textlint-plugin-text; do echo -n "$package: "; npm view $package deprecated; done
txt-to-ast: See https://github.com/textlint/textlint/issues/455
markdown-to-ast: See https://github.com/textlint/textlint/issues/455
textlint-ast-test: See https://github.com/textlint/textlint/issues/455
txt-ast-traverse: See https://github.com/textlint/textlint/issues/455
textlint-fixer-formatter: See https://github.com/textlint/textlint/issues/455
textlint-formatter: See https://github.com/textlint/textlint/issues/455
textlint-plugin-markdown: See https://github.com/textlint/textlint/issues/455
textlint-plugin-text: See https://github.com/textlint/textlint/issues/455
```
